app/controllers/companies_controller.rb

```diff
@@ -9,45 +12,33 @@ def new
 
   def create
     @company = Company.new(company_params)
-    if @company.save
-      flash[:success] = "#{@company.name} added!"
-      redirect_to company_path(@company)
-    else
-      render :new
-    end
+    create_company(@company)
   end
 
   def show
-    company = Company.find(params[:id])
-    redirect_to company_jobs_path(company)
+    redirect_to company_jobs_path(@company)
   end
 
   def edit
-    @company = Company.find(params[:id])
   end
 
   def update
-    @company = Company.find(params[:id])
     @company.update(company_params)
-    if @company.save
-      flash[:success] = "#{@company.name} updated!"
-      redirect_to company_path(@company)
-    else
-      render :edit
-    end
+    save(@company)
```

Comment:
provide consistency over your module. this was called `update_category` so would like to see this be `update_company`